contracts/TokenConverterRoute.sol

```diff
@@ -0,0 +1,64 @@
+pragma solidity ^0.4.24;
+
+import "./interfaces/TokenConverter.sol";
+import "./interfaces/AvailableProvider.sol";
+import "./interfaces/Token.sol";
+import "./utils/Ownable.sol";
+import "./vendors/bancor/converter/BancorGasPriceLimit.sol";
+
+
+contract TokenConverterRoute is TokenConverter, Ownable {
+    
+    uint256 constant internal MAX_UINT = uint256(0) - 1;
+    TokenConverter[] private converters;
+    mapping (address => address) private availability;
+    
+    function addConverter(TokenConverter converter, AvailableProvider availabilityContract) onlyOwner public {
+        converters.push(converter);
+        availability[converter] = availabilityContract;        
+    }
+    
+    function converter(Token _from, Token _to, uint256 _amount, uint256 _minReturn) external payable returns (uint256 amount) {
+        address betterProxy = _getBetterProxy(_from, _to, _amount);
+        TokenConverter converter =  TokenConverter(betterProxy);
+        return converter.convert.value(msg.value)(_from, _to, _amount, _minReturn);   
```

Comment:
Tokens should be pulled and then approved for this to work, the converter will try to make a token pull from this contract, and this contract didn't approve that.